alda-lang/alda
# Add an autoupdating script

Still a WIP, but the basic functions are working as of now as far as I know. This adds an `alda update` command that updates alda based off of the latest release. I've only tested this on linux, but I think this should work on windows/mac too (but it needs to be tested).

Currently, it simply overwrites the current alda executable with the one from the latest release (and does no checking to see if an update is needed or not). I still need to figure out how to do that (maybe starting the server to find the version). Another problem is that the version tags don't match up with the version reported by alda version (so getting the version from the server might not help too much either way).
    
As far as I know, I didn't need to make the downloaded file executable (It seemed to preserve the permissions of the old alda file? I'm going to double check this later).

Overview:
- Query GH Api to get latest release tag/download url
- Remove .exe if it exists from download url (for unix compatibility)
- Add .exe back to download URL if on windows
- Download release to path specified by current alda executbale
    
Let me know what feedback you have, and I'll get to fixing it!

Also, one more thing, something about installing launch4j should probably be added to the [developers guide](https://github.com/alda-lang/alda/blob/master/doc/development-guide.md#testing-changes).

Closes #82
